### PR TITLE
add warnings to existing Prow docs

### PR DESCRIPTION
**Pull request by listx**
Part of https://github.com/kubernetes-sigs/prow/issues/4.

This change just adds highly visible, almost obnoxious comments that are only visible to those who would like to make modifications to these files. The files now live in https://github.com/kubernetes-sigs/prow/pull/10 under the "Legacy Snapshot" subdirectory.

The next phase of this migration likely involves "promoting" each file out of the "Legacy Snapshot" directory there to a less "Legacy ..."-sounding path, and once that's done, deleting the original file here so as to force users to use docs.prow.k8s.io as the source of truth.

/assign @cjwagner

### CODE REVIEW

**Comment by cjwagner** [comment]
I think this is a perfect use case for the [`blockade` plugin](https://github.com/kubernetes/test-infra/blob/ad2eb087454f21458b679449ab3cb7f344eb43fa/prow/plugins/blockade/blockade.go#L17-L24). It allows us to block PRs from merging if they touch certain file paths. You can find a example config in the "details" [here](https://prow.k8s.io/plugins).

**Comment by listx** [comment]
> here

Indeed! Let me update this to use the plugin. I think I'll still leave the topmost comment in each file explaining why the file is "blocked", but otherwise remove the other "obnoxious" lines.

File: config/prow/plugins.yaml
Comment (cjwagner) [inline config/prow/plugins.yaml:254]: I suspect we'll want some exceptions. In particular we're definitely still going to want a README for Prow as a whole, even if it only really points to the documentation site. We might want other READMEs too? WDYT?

We can tweak this later so we don't need to determine this now.

**Review by cjwagner — APPROVED** [review]
/hold

File: config/prow/plugins.yaml
Comment (listx) [reply config/prow/plugins.yaml:254]: Exception for main Prow README sounds fine. I'm not sure which other ones need exceptions, so I'd rather leave that up to future PRs as they come up.

Comment (listx) [reply config/prow/plugins.yaml:256]: Added exception for `prow/README.md`.

**Pull request merged**
